oximeter/db/src/shells/native.rs

```diff
@@ -0,0 +1,194 @@
+    for _row in 0..n_rows {
+        let row = columns.iter_mut().map(|col_iter| col_iter.next().unwrap());
+        builder.push_record(row);
+    }
+    println!("{}", builder.build().with(Style::psql()));
+    println!();
+    println!("{}: {}", "Query ID".bold(), result.id);
+    println!("{}:  {:?}", "Elapsed".bold(), result.progress.query_time);
+}
+
+fn values_to_string<'a>(
+    values: &'a ValueArray,
+) -> Box<dyn Iterator<Item = String> + 'a> {
+    match values {
+        ValueArray::UInt8(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::UInt16(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::UInt32(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::UInt64(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::UInt128(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::Int8(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::Int16(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::Int32(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::Int64(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::Int128(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::Float32(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::Float64(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::String(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::Uuid(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::Ipv4(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::Ipv6(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::DateTime(vals) => {
+            Box::new(vals.iter().map(ToString::to_string))
+        }
+        ValueArray::DateTime64 { values, .. } => {
+            Box::new(values.iter().map(ToString::to_string))
+        }
+        ValueArray::Nullable { is_null, values } => {
+            let inner = values_to_string(values);
+            let it = is_null.iter().zip(inner).map(|(is_null, value)| {
+                if *is_null {
+                    String::from("NULL")
+                } else {
+                    value.to_string()
+                }
+            });
+            Box::new(it)
+        }
+        ValueArray::Enum8 { variants, values } => {
+            Box::new(values.iter().map(|i| variants.get(i).unwrap().clone()))
```

Comment:
`variants` is an `IndexMap<i8, String>`, So that gives the mapping from the variant "discriminant" to the string form of it. The `values` is a list of `i8`s, each of which is a key into `variants`, and gives the discriminant of the enum in that row.

So it's definitely true that `i` must exist in `variants`, which is why we `unwrap()` there, but they don't have to be the same length.